### ISSUE TITLE
Nested dialog folders aren't converted

### ISSUE THREAD

**Issue by tracyboehrer** [microsoft/BotFramework-Composer#7730]
## Describe the bug

When opening an old project Composer attempts to convert.  In cases where there are dialogs folders nested, the nested folder is ignored.

## Version

Version:  2.0.0-nightly.243450.6728c56
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Open the project in extensions/samples/assets/project/ActionsSample
2. Choose to convert the project, and start the bot
3. If you were to run the app and choose the "enddialog" action, and enter "joke", you'll get an error. (Note in JS, you'll get an error on load, but that appears to be a difference in how JS and C# load a bot).
4. You can see the difference is that in extensions\samples\assets\projects\ActionsSample\dialogs\enddialog, there is a dialogs folder that contains the "telljoke" dialog.  This isn't copied to the new project.  "replacedialog" also has a nested dialogs folder with two more dialogs ("telljokedialog", and "fortunetellerdialog").

## Expected behavior

All dialogs are copied.

## Other

There is still some funny behavior when I run this bot.  But it seems like something else.  Again, choose "05 end dialog".  I seem to be getting an error sometimes the first try.  Then it seems to recover.  Possible it's something on my end.

**Comment by srinaath** [comment]
@mewa1024  looking into the bug now.

**Comment by srinaath** [comment]
Thanks @tracyboehrer We are able to repro the error.

The old "ActionsSample" has a weird structure with nested dialogs. We had made a change based on  feedback to avoid tampering with user data and hence we wouldn't go search for nested dialog, lg, lu resources. The following are the glob patterns we look for. 
https://github.com/microsoft/BotFramework-Composer/pull/6041#discussion_r584480343

On 1.4.1, this works implicitly as the runtime picks those dialogs but not the Composer UI so it "looks" like it works there but you wont be able click through to the nested dialog. During, the migration as @benbrown  pointed out it follows the same patterns to copy and hence the nested dialog folder is missed causing the error.

I feel the change to the structure contributed to it and this ticket should be a noop. But, it is part of the old samples that we shipped which we dont maintain anymore. Shiproom can decide.

CC @cwhitten
